docroot/modules/custom/va_gov_backend/src/EventSubscriber/OrphansReportViewsEventSubscriber.php

```diff
@@ -0,0 +1,193 @@
+  }
+
+  /**
+   * Pre view event handler.
+   *
+   * @param \Drupal\views_event_dispatcher\Event\Views\ViewsPreViewEvent $event
+   *   The event.
+   */
+  public function preView(ViewsPreViewEvent $event): void {
+    $view = $event->getView();
+    if ($view->id() === 'orphaned_paragraphs') {
+      $types = $view->getHandlerTypes();
+      $orphan_count = 0;
+      $storage = $this->entityTypeManager->getStorage('paragraph');
+      $results = $storage->getQuery()->execute();
+      $result_chunks = array_chunk($results, 1000);
+      foreach ($result_chunks as $chunk) {
+        $paragraphs = $storage->loadMultiple($chunk);
+        /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+        foreach ($paragraphs as $paragraph) {
+          // $top_parent = va_gov_backend_get_top_parent_entity($paragraph);
+          $used = $this->purger->isUsed($paragraph);
+          if (!$used) {
+            $orphan_count++;
+          }
+        }
+      }
+      $var = [
+        '#markup' => '<p>' . $orphan_count . ' total orphaned paragraphs</p>',
+      ];
+      $markup = $this->renderer->renderRoot($var);
+      // Get the header handlers, and add our new one.
+      $headers = $view->getHandlers('header', 'orphaned_para_page');
+      $custom_header = [
+        'id' => 'area_text_custom',
+        'table' => 'views',
+        'field' => 'area_text_custom',
+        'relationship' => 'none',
+        'group_type' => 'group',
+        'admin_label' => '',
+        'empty' => '1',
+        'content' => $markup,
+        'plugin_id' => 'text_custom',
+        'weight' => -1,
+      ];
+      array_unshift($headers, $custom_header);
+
+      // Add the list of headers back in the right order.
+      $view->displayHandlers->get('orphaned_para_page')->setOption($types['header']['plural'], $headers);
+    }
+  }
+
+  /**
+   * Pre render event handler.
+   *
+   * @param \Drupal\views_event_dispatcher\Event\Views\ViewsPreRenderEvent $event
+   *   The event.
+   *
+   * @throws \Exception
+   */
+  public function preRender(ViewsPreRenderEvent $event): void {
+    $view = $event->getView();
+    if ($view->id() === 'orphaned_paragraphs') {
+      foreach ($view->result as $key => $value) {
+        /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+        $paragraph = $value->_entity;
+        $top_parent = va_gov_backend_get_top_parent_entity($paragraph);
+        $used = $this->purger->isUsed($paragraph);
+        if (!$used && !$top_parent) {
+          $link = $this->getTopParentEntityLink($paragraph);
+          $str = 'This is an orphan paragraph. ' . $link;
+          $value->_entity->set('revision_id', $str);
+        }
+        elseif (!$used && $top_parent && $top_parent->getEntityTypeId() !== 'paragraph') {
+          $str = 'This paragraph is connected to ' . $top_parent->toLink()->toString() . '. Moderation state is: ' . $top_parent->get('moderation_state')->getValue()[0]['value'];;
```

Comment:
This is an awesome touch.  Well done.
![image](https://user-images.githubusercontent.com/5752113/207623777-9473e55e-459b-4501-aabc-c6d6898a81b4.png)

A couple things on the code aspect of it: 
`get('moderation_state')->getValue()[0]['value']` is a risky grab because it assumes there is an element 0 and sub-element value.
this would have less risk
`get('moderation_state')->value`  It matches the pattern we are using elsewhere in our custom code.

Minor:
Multiple string concatenations are slower because it calls the string concatenation with every '.' it runs into.  Replacement is only called once, no matter how many replacements are present.  Personally I find it easier to read too.
```suggestion
          $str = "This paragraph is connected to {$top_parent->toLink()->toString()}. Moderation state is: {$top_parent->get('moderation_state')->value}.";
```